packages/headless/src/features/generated-answer/generated-answer-actions.ts

```diff
@@ -144,21 +144,46 @@ export const streamAnswer = createAsyncThunk<
   const abortController = extra.streamingClient?.streamGeneratedAnswer(
     request,
     {
-      write: (data: GeneratedAnswerStreamEventData) => {
-        dispatch(setIsLoading(false));
-        if (data.payload && data.payloadType) {
-          handleStreamPayload(data.payloadType, data.payload);
+      write: (data: GeneratedAnswerStreamEventData, streamId: string) => {
```

Comment:
I don't think it's necessary to pass `streamId` to these callbacks, since it's already part of `request`.

I'm also not sure whether it's necessary to pass `abortController` to these callbacks, since it looks like it's already returned by the function.

Lastly, in the `search-api-client`, we instantiate the `abortController` because the `SearchAPIClient` is responsible for aborting requests, but here the responsibility seems mixed between the action and the `GeneratedAnswerAPIClient`. I would recommend completely isolating the logic to one or the other.
* If you isolate the `AbortController` logic to `GeneratedAnswerAPIClient`, then you should ensure that `GeneratedAnswerAPIClient` stops calling `write` on old streams whenever a new stream is created. This would remove the need for these `if` conditions that compare `streamId` with `generativeQuestionAnsweringId`.
* If you isolate the `AbortController` logic to the action, then only the `signal` of the abort controller should be passed down to the stream. If any event such as `write`/`abort`/etc is called on a different streamId than the current context's, then the current context's `abortController.abort` should be called and the remaining logic of the callback should be skipped.

Reply:
Hey! thanks for the suggestions!
Removed the stream id from the callbacks parameters, now directly using the `request`.

I also kinda reduced the mixture in responsibility between `GeneratedAnswerAPIClient ` and the `GeneratedAnswerAction` on who will call the abort controller,  now only the API client is calling the abort controller and this abort controller is no longer passed to the callbacks.

I would also want to mention that if you think that further refactoring is needed, I think it would be best to address it in a new PR just because this one was opened specifically to solve the abortController issue  occurring in Salesforce when locker Service is enabled, an issue that needs to be solved ASAP cause a client is waiting for it.